### ISSUE TITLE
Sidekiq: Job arguments to CDMDEXER::TransformWorker do not serialize to JSON safely...

### ISSUE THREAD

**Issue by ewlarson** [UMNLibraries/cdm-blacklightify#16]
### Descriptive summary

Running the `bundle exec rake umedia:index:harvest` rake task returns a flurry of these deprecation warnings:

```
WARN: Job arguments to CDMDEXER::LoadWorker do not serialize to JSON safely. This will raise an error in
Sidekiq 7.0. See https://github.com/mperham/sidekiq/wiki/Best-Practices or raise an error today
by calling `Sidekiq.strict_args!` during Sidekiq initialization.
```
### Expected behavior

We should be able to run cleanly without deprecations. Investigate and resolve.

**Comment by mberkowski** [comment]
Most likely this is caused by symbolic keys passed to ETLWorker in its config hash. We need to dig through everywhere especially the solr url is accessed and allow it to work with a string rather than a symbol
```
   CDMDEXER::ETLWorker.perform_async(
      solr_config: { url: args.fetch(:solr_url) },
      oai_endpoint: args.fetch(:oai_endpoint),
      cdm_endpoint: args.fetch(:cdm_endpoint),
      set_spec: args[:set_spec] != '""' ? args[:set_spec] : nil,
      batch_size: args.fetch(:batch_size, 10),
      max_compounds: args.fetch(:max_compounds, 10)
    )
```

**Comment by mberkowski** [comment]
This also extends to `TransformWorker`, where it is passed a list of `field_mappings` which are not a pure hash. When loading those from yaml and passing them in via rails config settings, the array of mappings contains `Config` objects rather than a plain hash (as you get if you load from yaml manually) so a bit more is needed to cast it all to Hash and then stringify keys.

**Comment by mberkowski** [comment]
Fixed by 69bbf5e2be2f83765a9ba74094f9b7d30d387622
